### PR TITLE
Fix field plotting

### LINKED ISSUE #741: "compute_on_defer" not working with dask array
Hello all, 

I realized that 'fieldset.U.data' is now a dask array:
`dask.array<where, shape=(3, 28, 205, 223), dtype=float32, chunksize=(1, 28, 205, 223), chunktype=numpy.ndarray>`

This makes the simplest operation using 'compute_on_defer' not working:
```python
def compute(fieldset):
    for tind in fieldset.W[0].loaded_time_indices:
        fieldset.W[0].data[tind,0]=0
        fieldset.W[1].data[tind, 0] = fieldset.W[1].data[tind, 0].clip(min=0)
```
Result:
`NotImplementedError: Item assignment with <class 'tuple'> not supported`

Any ideas of how to go around this?

### PR DESCRIPTION
**Pull request by CKehl**
Added mini-test for loading and plotting NEMO data with dask for error replication and fix #741

Comment: little clean-up of open branches. This one is a small addition adding more stability to the plotting by checking that plotting works correctly. Valuable addition with little extra hassle - hence: merge-request.

### CODE REVIEW

File: parcels/scripts/plottrajectoriesfile.py [hunk 1]
```diff
@@ -71,7 +71,8 @@ def plotTrajectoriesFile(filename, mode='2d', tracerfile=None, tracerfield='P',
     if mode == '3d':
         from mpl_toolkits.mplot3d import Axes3D  # noqa
         plt.clf()  # clear the figure
-        ax = fig.gca(projection='3d')
+        # ax = fig.gca(projection='3d')
+        ax = fig.gca()
```
Comment (erikvansebille) [inline parcels/scripts/plottrajectoriesfile.py:75]: I don't understand this change. I think this is what breaks the unit-tests. The `ax` is now not set as 3D anymore, so there is no `set_zlevel` method. Revert back to original?

Comment (CKehl) [reply parcels/scripts/plottrajectoriesfile.py:75]: The change was necessary cause that line broke [bccb214](https://github.com/OceanParcels/parcels/pull/1247/commits/bccb2140274886f0ecbbd56dbfb019c036b2a3d1). In short: in Windows, the workflow obtains a newer version of matplotlib, and in matplotlib, the function `figure.gca()` has no `projection` parameters. Unexpected change, but therefore I needed to make this `projection` option part of the figure creation procedure. I guess help would be welcome from [bccb214](https://github.com/OceanParcels/parcels/pull/1247/commits/bccb2140274886f0ecbbd56dbfb019c036b2a3d1) onwards.

File: parcels/scripts/plottrajectoriesfile.py [hunk 2]
```diff
@@ -71,8 +71,7 @@ def plotTrajectoriesFile(filename, mode='2d', tracerfile=None, tracerfield='P',
     if mode == '3d':
         from mpl_toolkits.mplot3d import Axes3D  # noqa
         plt.clf()  # clear the figure
-        # ax = fig.gca(projection='3d')
-        ax = fig.gca()
+        ax = plt.axes(projection='3d')
```
Comment (CKehl) [inline parcels/scripts/plottrajectoriesfile.py:74]: nice

File: environment_py3_linux.yml [hunk 3]
```diff
@@ -28,4 +28,5 @@ dependencies:
   - nbval
   - scikit-learn
   - pykdtree
+  - cartopy
```
Comment (erikvansebille) [inline environment_py3_linux.yml:31]: Why is cartopy now need? That was not the case before, and I suggest not to keep it
```suggestion
```

File: environment_py3_win.yml [hunk 4]
```diff
@@ -19,10 +19,11 @@ dependencies:
...
   - tqdm
   - xarray>=0.5.1
-  - dask>=2.0
+  - dask<=2022.9.0
   - cftime>=1.3.1
   - ipykernel
   - pytest
   - nbval
   - pykdtree
+  - cartopy
```
Comment (erikvansebille) [inline environment_py3_win.yml:28]: See above, I suggest not to make cartopy an explicit dependency (as we had not done before)
```suggestion
```

File: parcels/plotting.py [hunk 5]
```diff
@@ -101,7 +102,7 @@ def plotparticles(particles, with_particles=True, show_time=None, field=None, do
 
 
 def plotfield(field, show_time=None, domain=None, depth_level=0, projection='PlateCarree', land=True,
-              vmin=None, vmax=None, savefile=None, **kwargs):
+              vmin=None, vmax=None, savefile=None, use3D=False, **kwargs):
```
Comment (erikvansebille) [inline parcels/plotting.py:105]: Is this variable still needed now? I am a bit confused what this does

File: parcels/plotting.py [hunk 6]
```diff
@@ -264,21 +268,26 @@ def plotfield(field, show_time=None, domain=None, depth_level=0, projection='Pla
...
+        ax.grid()
+    elif spherical and projection:
         try:
             import cartopy
         except:
             logger.info("Visualisation of field with geographic coordinates is not possible. Cartopy not found.")
             return None, None, None, None  # creating axes was not possible
 
         projection = cartopy.crs.PlateCarree(central_longitude) if projection == 'PlateCarree' else projection
+        # projection = '3d' if use3D else projection
```
Comment (erikvansebille) [inline parcels/plotting.py:290]: Stray debugging comment
```suggestion
```

File: tests/test_plotting.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+
+if __name__ == '__main__':
+    fset = test_field_from_netcdf()
+    print(fset)
+    pset = test_pset_create_field(fset)
+    kernels = pset.Kernel(AdvectionRK4) + periodicBC
+    pset.execute(kernels, dt=delta(hours=1), output_file=None,
+                 recovery={ErrorCode.ErrorOutOfBounds: DeleteParticle})
+    pset.show(field=fset.U)
```
Comment (erikvansebille) [inline tests/test_plotting.py:54]: What does this new set of functions test that was not tested before? Why is it needed?

File: environment_py3_linux.yml (hunk 3 above)
Comment (CKehl) [reply environment_py3_linux.yml:31]: that will make the _new_ test fail that actually tests platecarree plotting

File: parcels/plotting.py (hunk 5 above)
Comment (CKehl) [reply parcels/plotting.py:105]: possibly not, indeed

**Comment by erikvansebille** [comment]
I think that this was now fixed in 0326f773e0faee0cfb1400ad2bd0385833123936

**Pull request closed without merging**